### PR TITLE
Simplify iter8ctl usage and add assertions

### PR DESCRIPTION
**Pull request by sriumcp**
Partly addresses iter8-tools/iter8#850

### CODE REVIEW

File: experiment/experiment.go [hunk 1]
```diff
@@ -2,23 +2,165 @@
...
+	}, nil
+}
+
 // Started indicates if at least one iteration of the experiment has completed.
 func (e *Experiment) Started() bool {
+	if e == nil {
+		return false
+	}
 	c := e.Status.CompletedIterations
 	return c != nil && *c > 0
```
Comment (kalantar) [inline experiment/experiment.go:139]: Check stage instead?  `e.Status.Stage.After(v2alpha2.ExperimentStageInitializing)`

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -10,16 +10,26 @@ jobs:
   test:
     strategy:
       matrix:
-        go-version: [1.13.x, 1.15.x]
+        go-version: [1.16.x]
```
[the remark this answers is not in the document]
Comment (sriumcp) [reply .github/workflows/tests.yaml:13]: We are now testing only 1.16 (iter8ctl actually requires 1.16 now).

File: experiment/experiment.go (hunk 1 above)
Comment (sriumcp) [reply experiment/experiment.go:139]: Currently, `iter8ctl` description uses `started` to indicate at least one or more iterations of the experiment have been completed, so this check seems correct as it stands.

We can certainly revisit wording (for example, rename the `Started` function to `OneOrMoreIterationsOfTheExperimentCompleted`, to be accurate).

**Review by kalantar — COMMENTED** [review]
lgtm

**Pull request merged**